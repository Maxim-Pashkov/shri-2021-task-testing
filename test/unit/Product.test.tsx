import React from 'react';
import { it, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import { getApplication } from './getApplication';
import { Application } from '../../src/client/Application';

describe('Product test', () => {
    it('Product page loading is visible', () => {
        const { application, history } = getApplication(
            () => <Application />,
            {}
        );
        history.push('/catalog/10');

        const { container } = render(application);

        expect(container.querySelector('.Product')).toMatchSnapshot();
    });

    it('Product page is visible', async () => {
        const { application, history, store } = getApplication(
            () => <Application />,
            {}
        );
        history.push('/catalog/10');

        const { container } = render(application);

        await new Promise((resolve) => store.subscribe(() => resolve(true)));

        expect(container.querySelector('.Product')).toMatchSnapshot();
    });
});
