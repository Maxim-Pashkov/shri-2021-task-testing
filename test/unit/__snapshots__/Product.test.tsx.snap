// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product test Product page is visible 1`] = `
<div
  class="Product"
>
  <div
    class="ProductDetails row"
  >
    <div
      class="col-12 col-sm-5 col-lg-4"
    >
      <img
        class="Image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
      />
    </div>
    <div
      class="col-12 col-sm-7 col-lg-6"
    >
      <h1
        class="ProductDetails-Name"
      >
        test name
      </h1>
      <p
        class="ProductDetails-Description"
      >
        test description
      </p>
      <p
        class="ProductDetails-Price fs-3"
      >
        $
        120
      </p>
      <p>
        <button
          class="ProductDetails-AddToCart btn btn-primary btn-lg"
        >
          Add to Cart
        </button>
      </p>
      <dl>
        <dt>
          Color
        </dt>
        <dd
          class="ProductDetails-Color text-capitalize"
        >
          test color
        </dd>
        <dt>
          Material
        </dt>
        <dd
          class="ProductDetails-Material text-capitalize"
        >
          test material
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`Product test Product page loading is visible 1`] = `
<div
  class="Product"
>
  LOADING
</div>
`;

exports[`Product test Product page selected item is visible 1`] = `
<span
  class="CartBadge text-success mx-3"
>
  Item in cart
</span>
`;
